src/modules/job-info/job-info.c

```diff
@@ -128,6 +129,78 @@ static int lookup_key (struct lookup_ctx *l)
     return 0;
 }
 
+/* Parse the submit userid from the event log.
+ * Assume "submit" is the first event.
+ */
+static int eventlog_get_userid (struct lookup_ctx *l,
+                                const char *s, int *useridp)
+{
+    const char *input = s;
+    const char *tok;
+    size_t toklen;
+    char *event = NULL;
+    char name[FLUX_KVS_MAX_EVENT_NAME + 1];
+    char context[FLUX_KVS_MAX_EVENT_CONTEXT + 1];
+    json_t *o = NULL;
+
+    if (!eventlog_parse_next (&input, &tok, &toklen)) {
+        flux_log_error (l->ctx->h, "%s: invalid event", __FUNCTION__);
+        errno = EINVAL;
+        goto error;
+    }
+    if (!(event = strndup (tok, toklen)))
+        goto error;
+    if (flux_kvs_event_decode (event, NULL, name, sizeof (name),
+                               context, sizeof (context)) < 0)
+        goto error;
+    if (strcmp (name, "submit") != 0) {
+        flux_log_error (l->ctx->h, "%s: invalid event", __FUNCTION__);
+        errno = EINVAL;
+        goto error;
+    }
+    if (!(o = json_loads (context, 0, NULL))) {
+        errno = EPROTO;
+        goto error;
+    }
+    if (json_unpack (o, "{ s:i }", "userid", useridp) < 0) {
+        errno = EPROTO;
+        goto error;
+    }
+    free (event);
+    json_decref (o);
+    return 0;
+ error:
+    free (event);
```

Comment:
I think we need to save / restore errno in this block.

Reply:
ahhh yes, when we converted to json formatted contexts the json_decref was added.  Thanks!